api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3638,6 +3725,58 @@ private void skipStream(Object inputStream, long n)
   }
 
 
+  private int getAvailableSize(Object inputStream, int expectedReadSize) throws IOException, InternalException {
+    RandomAccessFile file = null;
+    BufferedInputStream stream = null;
+    if (inputStream instanceof RandomAccessFile) {
+      file = (RandomAccessFile) inputStream;
+    } else if (inputStream instanceof BufferedInputStream) {
+      stream = (BufferedInputStream) inputStream;
+    } else {
+      throw new InternalException("Unknown input stream. This should not happen.  Please report");
```

Comment:
> I don't think we should say "This should not happen" if the user can pass his input stream which is not RandomAccessFile or BufferedInputStream

Its not user passed input stream.  User passed input stream should have been wrapped with `BufferedInputStream` previously.  As its an internal error by programmers, the text is correct.